### ISSUE TITLE
Function to "squash" git commits

### ISSUE THREAD

**Issue by NixonInnes** [NixonInnes/Flask-Git#17]
Implement a function to consolidate several commits.

As far as I can tell there are two ways to achieve this; there's `git rebase` which most people seem to use in an interactive fashion (will have to research a single command line method);
or alternatively `git reset --soft HEAD~x` & `git commit -m 'new commit'` to squash the last x commits.
If anyone has any enlightenment to share, please do.

**Comment by tobywhughes** [comment]
Trying to figure out about how GitPython handles rebasing. There is nothing in the docs about it.

**Comment by NixonInnes** [comment]
Yeah, the documentation isn't very clear. You can use rebase like; 
```python
repo = git.Repo('/path/to/repo')
repo.git.rebase()
```
The `git` object gives access to a lot of the normal git functions.
